File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int maxHeight=0;
        int left=0;
        int right=heights.length-1;
        while(left<right){
            int leftHeight=heights[left];
            int rightHeight=heights[right];
            int currHeight=(right-left) * Math.min(leftHeight , rightHeight);
            maxHeight=Math.max(maxHeight,currHeight); 
            if(leftHeight<rightHeight) {
                left=left+1;
            }else{
                right=right-1;
            }
        }
        return maxHeight;
        
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left =0
        res=0
        count_map={}
        for right in range(len(s)):
            count_map[s[right]]=1+count_map.get(s[right],0)
            while right-left+1-max(count_map.values())>k:
                count_map[s[left]]-=1
                left=left+1
            res=max(res,right-left+1)

        return res
        





        